url088e9d87be_jpush_jpush_java_library_tgz-pJ8-cn_jpush_example_JPushClientExampleJ8/src/cn/jpush/api/ValidateRequestParams.java

```diff
@@ -0,0 +1,45 @@
+package cn.jpush.api;
+import java.util.regex.Pattern;
+
+public class ValidateRequestParams {
+	private final static int sendNo = -1;
+	private final static Pattern pattern = Pattern.compile("[^a-zA-Z0-9]");
+	
+	public static MessageResult vidateParams(MessageParams params){					
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 12-37 into a separate method.